ci-operator/step-registry/upi/install/gcp/upi-install-gcp-commands.sh

```diff
@@ -0,0 +1,461 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+export HOME=/tmp
+
+export SSH_PRIV_KEY_PATH=${CLUSTER_PROFILE_DIR}/ssh-privatekey
+export OPENSHIFT_INSTALL_INVOKER=openshift-internal-ci/${JOB_NAME_SAFE}/${BUILD_ID}
+
+echo "$(date -u --rfc-3339=seconds) - Configuring gcloud..."
+export GOOGLE_CLOUD_KEYFILE_JSON=${CLUSTER_PROFILE_DIR}/gce.json
+gcloud auth activate-service-account --key-file="${GOOGLE_CLOUD_KEYFILE_JSON}"
+gcloud config set project "$(jq -r .gcp.projectID ${SHARED_DIR}/metadata.json)"
+
+echo "$(date -u --rfc-3339=seconds) - Copying config from shared dir..."
+dir=/tmp/installer
+mkdir -p "${dir}/auth/"
+pushd ${dir}
+cp -t "${dir}" \
+    "${SHARED_DIR}/install-config.yaml" \
+    "${SHARED_DIR}/metadata.json" \
+    ${SHARED_DIR}/*.ign
+cp -t "${dir}/auth" \
+    "${SHARED_DIR}/kubeadmin-password" \
+    "${SHARED_DIR}/kubeconfig"
+cp -t "${dir}" \
+    /var/lib/openshift-install/upi/${CLUSTER_TYPE}/*
+tar -xzf "${SHARED_DIR}/.openshift_install_state.json.tgz"
+
+function backoff() {
+    local attempt=0
+    local failed=0
+    while true; do
+        "$@" && failed=0 || failed=1
+        if [[ $failed -eq 0 ]]; then
+            break
+        fi
+        attempt=$(( attempt + 1 ))
+        if [[ $attempt -gt 5 ]]; then
+            break
+        fi
+        echo "command failed, retrying in $(( 2 ** $attempt )) seconds"
+        sleep $(( 2 ** $attempt ))
+    done
+    return $failed
+}
+
+## Export variables to be used in examples below.
+echo "$(date -u --rfc-3339=seconds) - Exporting variables..."
+BASE_DOMAIN='origin-ci-int-gce.dev.openshift.com'
+BASE_DOMAIN_ZONE_NAME=$(gcloud dns managed-zones list --filter "DNS_NAME=${BASE_DOMAIN}." --format json | jq -r .[0].name)
+NETWORK_CIDR='10.0.0.0/16'
+MASTER_SUBNET_CIDR='10.0.0.0/19'
+WORKER_SUBNET_CIDR='10.0.32.0/19'
+
+KUBECONFIG="${dir}/auth/kubeconfig"
+export KUBECONFIG
```

Comment:
Does the installer care about `$KUBECONFIG`?  I thought it just used [the path][1].  Is there something besides the installer that is trying to consume this kubeconfig?

[1]: https://github.com/openshift/installer/blob/1a0bf0af2eed903d297c2221bea6f256c699004d/cmd/openshift-install/create.go#L95

Reply:
certificate approval.... `oc`